test/generators/suspenders/install/web_generator_test.rb

```diff
@@ -48,10 +48,34 @@ class WebGeneratorTest < Rails::Generators::TestCase
           end
         end
 
+        test "evaluates support for Node versions correctly" do
+          web_generator = Generators::Install::WebGenerator.new
+
+          unsupported_versions = %w[1.0.0 1.100.200 10.0.0 19.0.0 19.9.9 19.9999.99999]
+
+          unsupported_versions.each do |unsupported_version|
+            Generators::Install::WebGenerator.any_instance.stubs(:node_version).returns(unsupported_version)
+
+            assert_predicate web_generator, :node_version_unsupported?, "Node version #{unsupported_version} should not be supported"
+          end
+
+          supported_versions = %w[20.0.0 20.1.0 20.100.200 50.0.0 100.0.0]
+
+          supported_versions.each do |supported_version|
+            Generators::Install::WebGenerator.any_instance.stubs(:node_version).returns(supported_version)
+
+            assert_not_predicate web_generator, :node_version_unsupported?, "Node version #{supported_version} should be supported"
+          end
+        end
+
         private
 
         def prepare_destination
           touch "Gemfile"
+
+          File.open("test/dummy/Gemfile", "w") do |gemfile|
+            gemfile.write('source "https://rubygems.org"')
+          end
```

Comment:
While working in this file, if nothing was raised when a `run_generator` command ran, then there was a big slowdown and the following message would appear multiple times:

```
...[DEPRECATED] This Gemfile does not include an explicit global source. Not using an explicit global source may result in a different lockfile being generated depending on the gems you have installed locally before bundler is run. Instead, define a global source in your Gemfile like this: source "https://rubygems.org".
```

Reply:
Adding `source "https://rubygems.org"` to the `Gemfile` resolved this warning, making it easier to test the tests in the file.

Note that `run_generator` still fails for different reasons, but it's a faster fail with less noise.